### PR TITLE
Rails initialization

### PR DESCRIPTION
**Pull request by dLobatog**
I made this PR depend on https://github.com/Dynflow/dynflow/pull/216 - so let's try to get them in order to reduce the diff a little. You should be familiar with most of this code. Check out its usage on https://github.com/theforeman/foreman/pull/4316#discussion_r102676503

### CODE REVIEW

File: lib/dynflow.rb [hunk 1]
```diff
@@ -49,4 +49,20 @@ class Error < StandardError
   require 'dynflow/semaphores'
   require 'dynflow/throttle_limiter'
   require 'dynflow/config'
+
+  if defined? ::ActiveJob
+    require 'dynflow/active_job/queue_adapter'
+
+    class Railtie < Rails::Railtie
```
Comment (ohadlevy) [inline lib/dynflow.rb]: should this be moved into the rails section?

File: lib/dynflow/rails/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+      attr_accessor :lazy_initialization
+
+      # what rake tasks should run their own executor, not depending on the external one
+      attr_accessor :rake_tasks_with_executor
+
+      # If true, the ForemanTasks::Concerns::ActionTriggering will make
+      # no effect. Useful for testing, where we mignt not want to execute
+      # the orchestration tied to the models.
+      attr_accessor :disable_active_record_actions
```
Comment (iNecas) [inline lib/dynflow/rails/configuration.rb]: This is not valid for pure dynfow

File: lib/dynflow/rails/persistence.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+module Dynflow
```
Comment (iNecas) [inline lib/dynflow/rails/persistence.rb]: I don't think we need this file without foreman-tasks

File: lib/dynflow/rails/daemon.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        sleep
+      ensure
+        STDOUT.puts('Exiting')
+      end
+
+      # run the executor as a daemon
+      def run_background(command = 'start', options = {})
+        default_options = { rails_root: Dir.pwd,
+                            process_name: 'dynflow_executor',
+                            pid_dir: File.join(::Rails.root, '/tmp/pids'),
```
Comment (ohadlevy) [inline lib/dynflow/rails/daemon.rb]: I think it should be
```ruby
  pid_dir: File.join(::Rails.root, 'tmp','pids'),
```

File: lib/dynflow/rails/daemon.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                            log_dir: File.join(::Rails.root, 'log'),
+                            wait_attempts: 300,
+                            wait_sleep: 1,
+                            executors_count: (ENV['EXECUTORS_COUNT'] || 1).to_i }
+        options = default_options.merge(options)
+        FileUtils.mkdir_p(options[:pid_dir])
```
Comment (ohadlevy) [inline lib/dynflow/rails/daemon.rb]: not sure if its required, but i guess not harmful?

File: lib/dynflow/rails/configuration.rb (hunk 2 above)
Comment (dLobatog) [inline lib/dynflow/rails/configuration.rb]: OK - I suppose the tasks Configuration object could introduce it

File: lib/dynflow/rails/persistence.rb (hunk 3 above)
Comment (dLobatog) [inline lib/dynflow/rails/persistence.rb]: Indeed, I'll rewrite it in Foreman-tasks so it's Configuration file overrides initialize_persistence with one that actually saves stuff like this.

**Comment by dLobatog** [comment]
@iNecas Updated to remove the tasks option and the PersistenceAdapter for tasks. -tasks will simply override Configuration to define its own persistence model based on Sequel. Pretty similar to how Foreman does it: (https://github.com/theforeman/foreman/pull/4316/files#diff-b227cab047536b0261bee94c657af2deR2).

**Comment by dLobatog** [comment]
Updated to fix 1 small conflict with the ActiveJob provider 😄

**Comment by iNecas** [comment]
Thanks @dLobatog, I'm releasing new version now so that you can proceed with other parts of your effort

**Comment by iNecas** [comment]
dynflow-0.8.20 is all yours @dLobatog :)

**Pull request merged**